test/test_linear_space_lcs.py

```diff
@@ -0,0 +1,60 @@
+import os
+import unittest
+
+from approximate_string_matching import linear_space_lcs, lcs
+from generator import rand
+
+LCS_ALGORITHMS = [
+    linear_space_lcs.linear_space_lcs,
+]
+
+class TestLinearSpaceLongestCommonSubsequence(unittest.TestCase):
+  run_large = unittest.skipUnless(
+      os.environ.get('LARGE', False), 'Skip test in small runs')
+
+  """Test if substr is subsequence of the word"""
+  def is_subsequence(self, word, substr):
+    j = 0
+    for i in range(0, len(word)):
+      if word[i] == substr[j]:
+        j = j + 1
+      if j == len(substr):
+        return True
+
+    return False
+
+  """Check if lcs length equal to returned by needleman_wunsch algorithm
+      and if lcs is substring of both word given as arguments"""
+  def check_linear_space_lcs(self, t_1, t_2, n_1, n_2):
+    lcs_word = lcs.needleman_wunsch(t_1, t_2, n_1 - 1, n_2 - 1)
```

Comment:
If you want to use needleman_wunsch consistently with your examples, you have to invoke it like
    exact_word = lcs.needleman_wunsch('#' + t_1, '#' + t_2, n_1, n_2)[0]
    exact_length = len(exact_word)
Otherwise if you pass 'aaa' as t_1 it will ignore the first 'a' as it'd be on index 0 - and by convention all algorithms but yours are 1-indexed

Reply:
Now strings 1-indexed are accepted by linear_space_lcs algorithm so there is no more problem.